File: constraint_layout/app/src/main/java/com/example/constraint_layout/MainActivity.java
package com.example.constraint_layout;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.example.constraint_layout.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button relative_btn = (Button)findViewById(R.id.btn_relative);
        Button constraint_btn = (Button)findViewById(R.id.btn_constraint);


    }

    public void toRelative(View view){
        Intent intent = new Intent(getApplicationContext(), RelativeActivity.class);
        startActivity(intent);
    }

    public void toConstraint(View view){
        Intent intent = new Intent(getApplicationContext(), ConstraintActivity.class);
        startActivity(intent);
    }

}

File: udp_server_app/app/src/main/java/com/example/udp_server_app/ui/MainActivity.kt
package com.example.udp_server_app.ui

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.example.udp_server_app.CommonConst
import com.example.udp_server_app.R
import com.example.udp_server_app.ui.client.ClientActivity
import com.example.udp_server_app.ui.profile.ProfileFragment
import com.example.udp_server_app.ui.server.ServerActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.btn_set_profile).let { btnSetProfile ->
            btnSetProfile.setOnClickListener {
                addFrag(ProfileFragment(), CommonConst.PROFILE_FRAG)
            }
        }

        findViewById<Button>(R.id.btn_enter_room).let { btnEnterRoom ->
            btnEnterRoom.setOnClickListener {
                startActivity(Intent(this, ClientActivity::class.java))
            }
        }

        findViewById<Button>(R.id.btn_make_room).let { btnMakeRoom ->
            btnMakeRoom.setOnClickListener {
                startActivity(Intent(this, ServerActivity::class.java))
            }
        }
    }

    fun addFrag(frag: Fragment, tag: String) {
        supportFragmentManager.beginTransaction()
            .add(R.id.frag_container, frag, tag)
            .commitNow()
    }

    fun removeFrag(frag: Fragment) {
        supportFragmentManager.beginTransaction()
            .remove(frag)
            .commitNow()
    }
}